### DISCUSSION TITLE
Would be cool to have text file listing compilation options / flags

### DISCUSSION THREAD

**Discussion by Scr3amer** [jedisct1/libsodium#1276] [Ideas]
Hey guys,

I am on windows and I was trying to avoid using msys because it is super slow somehow in building this library plus I use some custom toolchain (Clang16 + UCRT + MinGW).

I know there is no plan to support Cmake, I don't mind writing my own script but it would be cool to have some reference on what are the various flags enabled for the various type of targets.

I digged in the .ac and .vcxproj to get all that info but I am sure I missed some of them. Here is what I found so far:

## Definitions:
SODIUM_STATIC
SODIUM_EXPORT=
WIN32
NDEBUG
_CONSOLE
UNICODE
_UNICODE
NATIVE_LITTLE_ENDIAN=1
_CRT_SECURE_NO_WARNINGS
HAVE_AMD64_ASM
CONFIGURED=1

What else do I need besides -Ofast -fomit-frame-pointer -m64 -march=westmere -mtune=generic ?

Would be cool to have a document explaining all defines and required compiler flags.

**Comment by jedisct1** [comment]
The purpose of the `configure` script is to check what works/what doesn't work on a given system, and define these constants and compiler flags accordingly.

I can send you the results for the current libsodium version, my CPU, my current macOS version, and my current compiler version, but that wouldn't be very useful.

I have no idea about the results on other systems, and this is going to change over time. Same for the set of checks being performed. When something possibly non-portable is added to the source code, a check is added to the configuration script. So that hopefully things will work even on systems I have zero visibility on.

Libsodium can be built with `zig`, btw. This is how all the Windows pre-built binaries for .NET are being created now. It's way faster than `mingw`.  Constants and flags are predefined instead of being tested, but that only works because what compiler and C library are being used.
